pkg/plugins/policies/core/rules/rules.go

```diff
@@ -301,8 +301,7 @@ func buildToList(p core_model.Resource, httpRoutes []core_model.Resource) ([]cor
 			for _, to := range policyWithTo.GetToList() {
 				rv = append(rv, &artificialPolicyItem{
 					targetRef: common_api.TargetRef{
-						Kind: common_api.MeshServiceSubset,
-						Name: mhrRules.TargetRef.Name,
```

Comment:
@lobkovilya This seems like it should depend on the `to` `targetRef`?

Reply:
You mean it should be the service of the `to targetRef` ? Maybe but then what happens when the `to targetRef` is `kind: Mesh` ?

Reply:
> Maybe but then what happens when the to targetRef is kind: Mesh ?

We don't support `kind: Mesh` today and we have an open issue https://github.com/kumahq/kuma/issues/5827. 

But even when we add `kind: Mesh` support in the future, we shouldn't lose `kuma.io/service` info, right? So at least it has to be included in `matchesHash`

Reply:
@lahabana in the interest of getting this change in for MeshGatewayRoute, should we make a note to implement this for MeshHTTPRoute separately?

Reply:
This doesn't support MeshGatewayRoute AFAIK. This adds matching for MeshGateways.

Reply:
Ok reverted this change as it should be handled as part of the MeshGateway support for MeshHTTPRoute